### PR TITLE
sched/sched_timerexpiration.c: change from sched_time to eventtime

### PR DESCRIPTION
**Pull request by zouboan**
## Summary
change from sched_time to eventtime to overcome the build error: 'struct sporadic_s' has no member named 'sched_time' , becuse the member to save  the last time that the scheduler ran is "eventtime" rather than "sched_time"
## Impact
sporadic sched
## Testing
ci

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@zouboan to avoid the break in the future, it's better to enable the sporadic policy in some config

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
@zouboan will you add a sample config (for `sim` or any other target) that enables sporadic policy in this PR? If yes, then I will postpone the merge of this PR until you add a config.

File: sched/sched/sched_timerexpiration.c [hunk 1]
```diff
@@ -205,8 +205,8 @@ static uint32_t nxsched_cpu_scheduler(int cpu, uint32_t ticks,
        * committed to updating the scheduler for this TCB.
        */
 
-      sporadic->sched_time.tv_sec  = g_sched_time.tv_sec;
-      sporadic->sched_time.tv_nsec = g_sched_time.tv_nsec;
+      sporadic->eventtime = USEC2TICK(1000000 * (uint64_t)g_sched_time.tv_sec
```
Comment (pkarashchenko) [inline sched/sched/sched_timerexpiration.c:208]:
```suggestion
      sporadic->eventtime = USEC2TICK(USEC_PER_SEC * (uint64_t)g_sched_time.tv_sec
```

Comment (pkarashchenko) [reply sched/sched/sched_timerexpiration.c:208]: Or maybe even use `sporadic->eventtime = SEC2TICK(g_sched_time.tv_sec) + NSEC2TICK(g_sched_time.tv_nsec)`?

Comment (zouboan) [reply sched/sched/sched_timerexpiration.c]: I'm so sorry for my mistake,thanks very much for your help!

File: sched/sched/sched_timerexpiration.c [hunk 2]
```diff
@@ -205,8 +205,8 @@ static uint32_t nxsched_cpu_scheduler(int cpu, uint32_t ticks,
        * committed to updating the scheduler for this TCB.
        */
 
-      sporadic->sched_time.tv_sec  = g_sched_time.tv_sec;
-      sporadic->sched_time.tv_nsec = g_sched_time.tv_nsec;
+      sporadic->eventtime = SEC2TICK(g_sched_time.tv_sec) +
+      NSEC2TICK(g_sched_time.tv_nsec);
```
Comment (pkarashchenko) [inline sched/sched/sched_timerexpiration.c:209]:
```suggestion
      sporadic->eventtime = SEC2TICK(g_sched_time.tv_sec) +
                            NSEC2TICK(g_sched_time.tv_nsec);
```

**Review by pkarashchenko — COMMENTED** [review]
Only one little style change.

**Comment by pkarashchenko** [comment]
Please squash to a single commit.

**Comment by zouboan** [comment]
@pkarashchenko I enabled sporadic sched in sim:foc , because i have tesed sporadic sched in my custom board about foc

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Pull request merged**